Implement key and mouse related events and also fix evil craft

AlphaMode opened the pull request:
EvilCraft was so cursed to get working

Also fixes lightmaps on forge models

AlphaMode committed:
EvilCraft works now

AlphaMode committed:
Implement keybind related events

AlphaMode committed:
Fix scroll event

AlphaMode committed:
Change BlockInitTrackerMixin to just catch the error

BluSpring commented on compat/forge-sodium-compats/src/main/java/xyz/bluspring/kilt/compat/sodium/mixin/cyclopscore/ModBaseMixin.java:
```diff
@@ -0,0 +1,16 @@
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+
+@Mixin(value = ModBase.class, remap = false)
```
You can use targets and not require a dependency here.

BluSpring commented on compat/forge-sodium-compats/src/main/java/xyz/bluspring/kilt/compat/sodium/mixin/cyclopscore/MinecraftHelpersMixin.java:
```diff
@@ -0,0 +1,17 @@
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Overwrite;
+
+@Mixin(value = MinecraftHelpers.class, remap = false)
```
You can use targets and not require a dependency here.

BluSpring commented on src/main/java/xyz/bluspring/kilt/forgeinjects/blaze3d/vertex/VertexConsumerInject.java:
```diff
@@ -26,6 +31,12 @@ default void putBulkData(PoseStack.Pose pose, BakedQuad bakedQuad, float[] fs, f
     }
 
+    // Most certainly going to break in 1.21.1 so beware
+    @ModifyVariable(method = "putBulkData(Lcom/mojang/blaze3d/vertex/PoseStack$Pose;Lnet/minecraft/client/renderer/block/model/BakedQuad;[FFFF[IIZ)V", at = @At("STORE"), index = 28, name = "r")
```
Avoid using index or name, switch to ordinal.

BluSpring reviewed (changes requested):
fix this lmao

AlphaMode committed:
Resolve PR issues
Stop depending on cyclopscore
Change VertexConsumerInject#kilt$applyLightmap to use ordinal for targeting

BluSpring commented:
Looks perfect!

AlphaMode commented:
Wait
